chasm/lib/activity/activity_tasks.go

```diff
@@ -0,0 +1,76 @@
+package activity
+
+import (
+	"context"
+
+	"go.temporal.io/server/api/matchingservice/v1"
+	"go.temporal.io/server/chasm"
+	"go.temporal.io/server/chasm/lib/activity/gen/activitypb/v1"
+	"go.temporal.io/server/common/resource"
+	"go.uber.org/fx"
+)
+
+type ActivityStartTaskExecutorOptions struct {
+	fx.In
+
+	MatchingClient resource.MatchingClient
+}
+
+type ActivityStartTaskExecutor struct {
+	MatchingClient resource.MatchingClient
```

Comment:
nit: you can just embed the options and you'll have everything available directly to the executor.